src/Table/Table.jsx

```diff
@@ -0,0 +1,132 @@
+    };
+  }
+
+  componentWillMount() {
+    if (this.props.initialSortState) {
+      this._sort(this.props.initialSortState);
+    }
+  }
+
+  _toggleSort(columnIndex) {
+    const {sortedData, sortState} = this.state;
+
+    if (sortState && sortState.columnIndex === columnIndex) {
+      sortState.direction =
+        sortState.direction === sortDirection.ASCENDING ? sortDirection.DESCENDING : sortDirection.ASCENDING;
+      this.setState({sortState, sortedData: sortedData.reverse()});
+      this.props.onSortChange(sortState);
+      return;
+    }
+
+    this._sort({
+      columnIndex,
+      direction: sortDirection.ASCENDING,
+    });
+  }
+
+  _sort(sortState) {
+    const columns = this.props.children;
+
+    if (typeof sortState.columnIndex !== "number" && sortState.columnID) {
+      sortState.columnIndex = lodash.findIndex(columns, column => column.props.id === sortState.columnID);
+    }
+
+    const sortedColumn = this.props.children[sortState.columnIndex];
+    if (!sortedColumn || !sortedColumn.props.sortable) {
+      return;
+    }
+
+    let sortedData = lodash(this.props.data)
+      .sortBy(row => {
+        let value = sortedColumn.props.sortValueFn(row);
+
+        if (typeof value === "string") {
+          value = value.trim().toLowerCase();
+        }
+
+        return value;
+      });
+
+    if (sortState.direction === sortDirection.DESCENDING) {
+      sortedData = sortedData.reverse();
+    }
+
+    sortState.columnID = sortedColumn.props.id;
+    this.setState({sortState, sortedData: sortedData.value()});
+    this.props.onSortChange(sortState);
+  }
+
+  render() {
+    const {children, className, filter, fixed, rowIDFn} = this.props;
+    const {sortedData, sortState} = this.state;
+    const {cssClass} = Table;
+
+    return (
+      <table className={joinClasses(cssClass.TABLE, fixed && cssClass.FIXED, className)}>
+        <Header onSortChange={columnIndex => this._toggleSort(columnIndex)} sortState={sortState}>
+          {children}
+        </Header>
+        <tbody className={cssClass.BODY}>
+          {sortedData.map((rowData, rowIndex) => (filter(rowData) ? (
+            <tr className={cssClass.ROW} key={rowIDFn ? rowIDFn(rowData) : rowIndex}>
+              {children.map(({props: col}, colIndex) => (
+                <Cell className={col.cell.className} key={col.id || colIndex} noWrap={col.noWrap}>
+                  {col.cell.renderer(rowData)}
+                </Cell>
+              ))}
+            </tr>
+          ) : null))}
+        </tbody>
+      </table>
+      // TODO(kofi): Add pagination footer or maybe a PagingTable wrapper.
+    );
+  }
+}
+
+Table.propTypes = {
+  children: PropTypes.arrayOf(PropTypes.shape({
+    type: PropTypes.oneOf([Column]),
+  })),
+  className: PropTypes.string,
+  data: PropTypes.array.isRequired,
+  filter: PropTypes.func,
+  fixed: PropTypes.bool,
+  initialSortState: tablePropTypes.sortState,
+  onSortChange: PropTypes.func,
+  rowIDFn: PropTypes.func,
```

Comment:
I don't think we have a convention around this but I prefer camelcasing acronyms (i.e. `rowIdFn`) just to disambiguate with consecutive acronyms, and for consistency. minor nit